File: dotnet-desktop-guide/net/winforms/input-mouse/snippets/how-to-manage-cursor-pointer/vb/Form1.vb
Imports System.Windows.Forms
Imports System.Drawing

Partial Public Class Form1

    Public Sub New()
        InitializeComponent()
    End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        '<SetControlCursor>
        Button2.Cursor = System.Windows.Forms.Cursors.Hand
        '</SetControlCursor>
    End Sub

    '<MoveCursor>
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        PointToScreen(Button2.Location)
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        PointToScreen(Button1.Location)
    End Sub
    '</MoveCursor>

    '<ShowHideCursor>
    Private Sub Button1_MouseEnter(sender As Object, e As EventArgs) Handles Button1.MouseEnter
        Cursor.Hide()
    End Sub

    Private Sub Button1_MouseLeave(sender As Object, e As EventArgs) Handles Button1.MouseLeave
        Cursor.Show()
    End Sub
    '</ShowHideCursor>

End Class

File: dotnet-desktop-guide/winforms/input-mouse/snippets/how-to-manage-cursor-pointer/csharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Runtime.InteropServices;

namespace project
{
    public partial class Form1 : Form
    {

        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            //<SetControlCursor>
            button2.Cursor = System.Windows.Forms.Cursors.Hand;
            //</SetControlCursor>
        }

        //<MoveCursor>
        private void button1_Click(object sender, EventArgs e) =>
            Cursor.Position = PointToScreen(button2.Location);

        private void button2_Click(object sender, EventArgs e) =>
            Cursor.Position = PointToScreen(button1.Location);
        //</MoveCursor>

        //<ShowHideCursor>
        private void button1_MouseEnter(object sender, EventArgs e) =>
            Cursor.Hide();

        private void button1_MouseLeave(object sender, EventArgs e) =>
            Cursor.Show();
        //</ShowHideCursor>
    }
}
